File: app/src/main/java/com/tebet/mojual/view/sale/SaleFragmentProvider.java
package com.tebet.mojual.view.sale;

import com.tebet.mojual.view.salenow.SaleNowFragment;
import dagger.Module;
import dagger.android.ContributesAndroidInjector;

@Module
public abstract class SaleFragmentProvider {

    @ContributesAndroidInjector(modules = SaleFragmentModule.class)
    abstract SaleFragment provideSaleFragmentFactory();

    @ContributesAndroidInjector(modules = SaleFragmentModule.class)
    abstract SaleNowFragment provideSaleNowFragmentFactory();
}

File: app/src/main/java/com/tebet/mojual/view/sale/SaleFragmentProvider.kt
package com.tebet.mojual.view.sale

import dagger.Module
import dagger.android.ContributesAndroidInjector

@Module
abstract class SaleFragmentProvider {

    @ContributesAndroidInjector(modules = [SaleFragmentModule::class])
    internal abstract fun provideSaleFragmentFactory(): SaleFragment
}
